lldb/source/Plugins/Language/Swift/SwiftMangled.h

```diff
@@ -0,0 +1,94 @@
+//===-- SwiftMangled.h ------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef liblldb_SwiftMangled_h_
+#define liblldb_SwiftMangled_h_
+
+#include "lldb/Core/DemangledNameInfo.h"
+#include "swift/Demangling/Demangle.h"
+
+using namespace swift::Demangle;
+
+/// A NodePrinter class with range tracking capabilities.
+///
+/// When used instead of a regular NodePrinter, this class will store additional
+/// range information of the demangled name in the `info` attribute, such as the
+/// range of the name of a method.
+class TrackingNodePrinter : public NodePrinter {
+public:
+  TrackingNodePrinter(DemangleOptions options) : NodePrinter(options) {}
+
+  lldb_private::DemangledNameInfo takeInfo() { return std::move(info); }
```

Comment:
The `std::move` would leave the `TrackingNodePrinter` in an unspecified state. Is that intended? Based on the usage I see we never use the `TrackingNodePrinter` after we call `takeInfo`, so maybe that's fine. But realistically, moving versus copying `DemangledNameInfo` should be equivalent. So I think we can leave out the `std::move` here.

Reply:
I renamed the method to `getInfo` as well.